pkg/api/types.go

```diff
@@ -2143,6 +2143,20 @@ const (
 
 	// SSHAuthPrivateKey is the key of the required SSH private key for SecretTypeSSHAuth secrets
 	SSHAuthPrivateKey = "ssh-privatekey"
+
+	// SecretTypeTLS contains information about SSL termination used with the Ingress resource.
+	//
+	// Required fields:
+	// - Secret.Data["key"] - TLS private key.
+	//   Secret.Data["cert"] - TLS certificate.
+	//   Secret.Data["chain"] - optional certificate chain.
+	// TODO: Consider supporting different formats, specifying CA/destinationCA.
+	SecretTypeTLS SecretType = "kubernetes.io/tls"
+
+	// TLSCertKey is the key for tls certificates in a TLS secert.
+	TLSCertKey = "cert"
```

Comment:
Should this be "tls.crt"? or something similar, so that the default name on the filesystem is right?  Most keys are "NAME.crt", "NAME.key".  We won't know the name up front, so do we want to be "ARBITRARYNAME.key", ".key", or "key"?